### DISCUSSION TITLE
Why is the tunneling direction reversed in the Floquet Hamiltonian?

### DISCUSSION THREAD

**Discussion by Wihtedeka** [QuSpin/QuSpin#500] [Q&A]
Hi, I've been trying work out a Floquet Hamiltonian for a specific effective Hamiltonian and I want to simulate what happens when using the Floquet Hamiltonian instead of just the first order static Hamiltonian from the Floquet-Magnus expansion. However when using the 'Floquet' class to compute the Floquet Hamiltonian I didn't get the results that I was expecting (there might very well be a mistake in my calculations). Trying out some minimal examples I let the Floquet class compute the Floquet Hamiltonian for a simple 1D bosonic chain. The operators I implemented are hopping to the right with time oscillating phases. 
```
from quspin.basis import boson_basis_1d 
from quspin.operators import hamiltonian
from quspin.tools.Floquet import Floquet 
import numpy as np 
pi = np.pi 

N = 5 
basis = boson_basis_1d(N, 1)
omega = 100 
T = 2*pi/omega 

dyn = [['+-', [[-1, l+1, l]], lambda t: np.exp(1j*np.cos(omega*t)), []] for l in range(N-1)]
Hper = hamiltonian([], dyn, basis=basis, check_herm=False, check_pcon=False, check_symm=False)

Floq = Floquet({'H': Hper, 'T': T}, HF=True)

print(np.round(Hper.toarray()))
print(np.round(Floq.HF))
```
In the Floquet Hamiltonian I only see matrix elements corresponding to tunneling to the left. Have I made a mistake in my reasoning? Is this about the micromotion operator that I've neglected so far? Or does the numerical calculation somehow assume a hermitian Hamiltonian? (However this same sort of transposition also happens in the more complicated model that I'm trying to analyse.)

**Comment by mgbukov** [comment]
if you turn on the hermiticity check `check_herm=True`, quspin will yell at you that your `Hper` is not hermitian. You need to add manually the '-+' term ;)

**Reply by Wihtedeka** [reply]
Okay, thank you. I probably have a mistake somewhere in my calculations/code then, I'll keep looking.
